### PR TITLE
httplab 0.2.1 (new formula)

### PR DESCRIPTION
**Pull request by m1ome**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/httplab.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/gchaincl/httplab").install buildpath.children
+    cd "src/github.com/gchaincl/httplab" do
+      system "go", "build", "-o", bin/"httplab"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    system "#{bin}/httplab", "-version"
```
Comment (bfontaine) [inline Formula/httplab.rb]: Can’t we write a test that does something more substantial?

Comment (m1ome) [inline Formula/httplab.rb]: @bfontaine a lot of formula doing such type of test, just checking that tool installed and echoe version

Comment (dunn) [inline Formula/httplab.rb]: @m1ome that's only when we're unable to run a more comprehensive test.

**Comment by ilovezfs** [comment]
@m1ome Thanks for the PR. Please feel free to reopen when the requested changes have been made.

File: Formula/httplab.rb (hunk 1 above)
Comment (gchaincl) [inline Formula/httplab.rb]: @dunn can u give me an example of a more comprehensive test?

Comment (OJFord) [reply Formula/httplab.rb]: I'd like to see this merged; I'm happy to help if I can, but I'm inclined to agree that I find it hard to imagine what a good test would look like here: httplab is interactive by design.

Comment (gchaincl) [reply Formula/httplab.rb]: there are some unit test, is this about running them?

**Comment by gchaincl** [comment]
@ilovezfs as @OJFord mentioned above, HTTPLab is interactive by design, so how should a test look like?

**Comment by gchaincl** [comment]
so if I understand correctly, `-version` would be enough if as part of the script we verify the output?

**Comment by ilovezfs** [comment]
@gchaincl no it needs to test something with the functionality. That specifically means not a --version or --help test.

**Comment by gchaincl** [comment]
ok, do you know any interactive program which does that, so I can get an example, I can't figure out how to test it

**Comment by gchaincl** [comment]
I think that would be enough to take a look, thanks!

File: Formula/httplab.rb (hunk 1 above)
Comment (gchaincl) [reply Formula/httplab.rb]: Something like this should do it:
```ruby
 (testpath/"test.exp").write <<~EOS
        spawn #{bin}/httplab
        sleep 1
        send \003
        expect eof
      EOS
      system "expect", "-f", "text.exp"
end
```

**Pull request closed without merging**